src/lib.rs

```diff
@@ -49,33 +49,53 @@ use report::{Method, Report};
 
 use std::borrow::Cow;
 use std::io::Result as IoResult;
+use std::marker::PhantomData;
 use std::panic::PanicInfo;
 use std::path::{Path, PathBuf};
 
+pub type MaybeString = Option<Cow<'static, str>>;
+
 /// A convenient metadata struct that describes a crate
 ///
 /// See [`metadata!`]
 pub struct Metadata {
     /// The crate version
-    pub version: Cow<'static, str>,
+    pub version: MaybeString,
     /// The crate name
-    pub name: Cow<'static, str>,
+    pub name: MaybeString,
     /// The list of authors of the crate
-    pub authors: Cow<'static, str>,
+    pub authors: MaybeString,
     /// The URL of the crate's website
-    pub homepage: Cow<'static, str>,
+    pub homepage: MaybeString,
+    /// The support information
+    pub support: MaybeString,
+
+    // prevent initialized in literal syntax
+    phantom_data: PhantomData<()>,
 }
 
-/// Initialize [`Metadata`]
+impl Default for Metadata {
```

Comment:
User should always use `metadata!` as the default. But such a constructor method must be `pub` to let `metadata!` work.

Maybe impl with a name like `__never_make_metadata_directly()`.